util/dvsim/FlowCfg.py

```diff
@@ -210,12 +230,69 @@ def resolve_hjson_raw(self, hjson_dict):
 
         # Parse master cfg files
         if self.is_master_cfg:
-            for cfg_file in use_cfgs:
-                # Substitute wildcards in cfg_file files since we need to process
-                # them right away.
-                cfg_file = subst_wildcards(cfg_file, self.__dict__)
-                self.cfgs.append(
-                    self.create_instance(cfg_file, self.proj_root, self.args))
+            for entry in use_cfgs:
+                if type(entry) is str:
+                    # Treat this as a file entry
+                    # Substitute wildcards in cfg_file files since we need to process
+                    # them right away.
+                    cfg_file = subst_wildcards(entry, self.__dict__)
+                    self.cfgs.append(
+                        self.create_instance(cfg_file, self.proj_root,
+                                             self.args))
+
+                elif type(entry) is dict:
+                    # Treat this as a cfg expanded in-line
+                    temp_cfg_file = self._conv_inline_cfg_to_hjson(entry)
+                    if not temp_cfg_file: continue
+                    self.cfgs.append(
+                        self.create_instance(temp_cfg_file, self.proj_root,
+                                             self.args))
+                    self.temp_cfg_files.append(temp_cfg_file)
+
+                else:
+                    log.error(
+                        "Type of entry \"%s\" in the \"use_cfgs\" key is invalid: %s",
+                        entry, str(type(entry)))
+                    sys.exit(1)
+
+    def _conv_inline_cfg_to_hjson(self, idict):
+        '''Dump a temp hjson file in the scratch space from input dict.
+        This method is to be called only by a master cfg'''
+
+        if not self.is_master_cfg:
+            log.fatal("This method can only be called by a master cfg")
+            sys.exit(1)
+
+        name = idict["name"] if "name" in idict.keys() else None
+        if not name:
+            log.error(
+                "In-line entry in use_cfgs list does not contain \
+                      a \"name\" key (will be skipped!):\n%s", idict)
+            return None
+
+        # Check if temp cfg file already exists
+        temp_cfg_file = self.scratch_root + "/." + self.branch + "__" + \
+                        name + "_cfg.hjson"
```

Comment:
not sure, but doesn't this lead to result path conflicts in the publish functions later on? 

say, if we were to use this inline mechanism for multiple different flows like synthesis and lint, wouldn't the summaries overwrite each other since they live in the same base path?

Reply:
The publish functions use `rel_path`, which is constructed in `__init__` from the path to this file. Since this is a 'fake' file that is generated to support this flow, we will need to move `rel_path` to the hjson instead. Let me think more on this.

Reply:
I made some more changes. Inline cfgs can now specify the `rel_path` to indicate where to publish the final results.